packages/interaction-tracking/src/InteractionTracking.js

```diff
@@ -0,0 +1,306 @@
+// Meaning that newly tracked interactions are appended to the previously active set.
+// When an interaction goes out of scope, the previous set (if any) is restored.
+let interactionsRef: InteractionsRef = (null: any);
+
+// Listener(s) to notify when interactions begin and end.
+// Note that subscribers are only supported when enableInteractionTrackingObserver is enabled.
+let subscriberRef: SubscriberRef = (null: any);
+
+if (enableInteractionTracking) {
+  interactionsRef = {
+    current: new Set(),
+  };
+  if (enableInteractionTrackingObserver) {
+    subscriberRef = {
+      current: null,
+    };
+  }
+}
+
+// These values are exported for libraries with advanced use cases (i.e. React).
+// They should not typically be accessed directly.
+export {interactionsRef as __interactionsRef, subscriberRef as __subscriberRef};
+
+export function clear(callback: Function): any {
+  if (!enableInteractionTracking) {
+    return callback();
+  }
+
+  const prevInteractions = interactionsRef.current;
+  interactionsRef.current = new Set();
+
+  try {
+    return callback();
+  } finally {
+    interactionsRef.current = prevInteractions;
+  }
+}
+
+export function getCurrent(): Set<Interaction> | null {
+  if (!enableInteractionTracking) {
+    return null;
+  } else {
+    return interactionsRef.current;
+  }
+}
+
+export function getThreadID(): number {
+  return ++threadIDCounter;
+}
+
+export function subscribe(subscriber: Subscriber): void {
+  if (enableInteractionTracking && enableInteractionTrackingObserver) {
+    invariant(
+      subscriberRef.current === null || subscriberRef.current === subscriber,
+      'Only one interactions subscriber may be registered at a time.',
+    );
+    subscriberRef.current = subscriber;
+  }
+}
+
+export function track(
+  name: string,
+  timestamp: number,
+  callback: Function,
+  threadID: number = DEFAULT_THREAD_ID,
+): any {
+  if (!enableInteractionTracking) {
+    return callback();
+  }
+
+  const interaction: Interaction = {
+    __count: 0,
+    id: interactionIDCounter++,
+    name,
+    timestamp,
+  };
+
+  const prevInteractions = interactionsRef.current;
+
+  // Tracked interactions should stack/accumulate.
+  // To do that, clone the current interactions.
+  // The previous set will be restored upon completion.
+  const interactions = new Set(prevInteractions);
+  interactions.add(interaction);
+  interactionsRef.current = interactions;
+
+  if (enableInteractionTrackingObserver) {
+    // Update before calling callback in case it schedules follow-up work.
+    interaction.__count = 1;
+
+    let returnValue;
+    const subscriber = subscriberRef.current;
+
+    try {
+      if (subscriber !== null) {
+        subscriber.onInteractionTracked(interaction);
+      }
+    } finally {
+      try {
+        if (subscriber !== null) {
+          subscriber.onWorkStarted(interactions, threadID);
+        }
+      } finally {
+        try {
+          returnValue = callback();
```

Comment:
You can actually just return here. The finally will still execute before returning.

Reply:
Returning in finally masks thrown errors.